api/src/entities/collection_mints.rs

```diff
@@ -70,8 +70,16 @@ impl CollectionMint {
             metadata_json_loader,
             ..
         } = ctx.data::<AppContext>()?;
+        let collection = self.collection(ctx).await?.ok_or(Error::new(format!(
+            "Collection not found for collection mint {:?}",
+            &self.id
+        )))?;
 
-        metadata_json_loader.load_one(self.id).await
+        if collection.blockchain == Blockchain::Solana {
+            metadata_json_loader.load_one(self.id).await
+        } else {
+            metadata_json_loader.load_one(self.collection_id).await
+        }
```

Comment:
lets do a match statement so when we add more blockchains the compiler reminds us to review this.